controllers/comments.controller.js

```diff
@@ -17,7 +19,10 @@ exports.postCommentByArticleId = (req, res, next) => {
     const { article_id } = req.params;
     const { username, body } = req.body;
 
-    insertCommentByArticleId(article_id, username, body)
+    Promise.all([checkUserExists(username), checkArticleExists(article_id)])
+        .then(() => {
+            return insertCommentByArticleId(article_id, username, body);
+        })
```

Comment:
Great use of Promise.all() and util function check...Exists
checkUserExists and checkArticleExists can be more DRY. Instead of creating 2 util function, can combine these two. Take a look at advanced error handling in NCNotes

Reply:
Actually checkArticleExists can be replaced by selectArticleById